src/dataclay/runtime/backend.py

```diff
@@ -20,95 +17,98 @@
+        await self.backend_clients.stop()
 
-    # Shutdown
-    def stop(self):
         # Remove backend entry from metadata
-        self.metadata_service.delete_backend(self.backend_id)
+        await self.metadata_service.delete_backend(self.backend_id)
 
-        # Stop DataManager
-        logger.debug("Stopping DataManager")
-        self.data_manager.stop()
-        self.data_manager.join()
-        logger.debug("DataManager stopped.")
+        # Stop DataManager memory monitor
+        self.data_manager.stop_memory_monitor()
 
-        self.close_backend_clients()
+        # Flush all data if not ephemeral
         if not settings.ephemeral:
-            self.data_manager.flush_all()
-
-    # Replicas
-
-    def synchronize(self, instance, operation_name, params):
-        raise Exception("To refactor")
-        session_id = self.session.id
-        object_id = instance._dc_meta.id
-        operation = self.get_operation_info(instance._dc_meta.id, operation_name)
-        implementation_id = self.get_implementation_id(instance._dc_meta.id, operation_name)
-        # === SERIALIZE PARAMETERS ===
-        serialized_params = SerializationLibUtilsSingleton.serialize_params_or_return(
-            params=[params],
-            iface_bitmaps=None,
-            params_spec=operation.params,
-            params_order=operation.paramsOrder,
-            hint_volatiles=instance._dc_meta.master_backend_id,
-            runtime=self,
-        )
-        self.execution_environment.synchronize(
-            session_id, object_id, implementation_id, serialized_params
-        )
-
-    # Federations
-
-    def federate_to_backend(self, dc_obj, external_execution_environment_id, recursive):
-        raise Exception("To refactor")
-        object_id = dc_obj._dc_meta.id
-        session_id = self.session.id
-        logger.debug(
-            "[==FederateObject==] Starting federation of object by %s with dest dataClay %s, and session %s",
-            object_id,
-            external_execution_environment_id,
-            session_id,
-        )
-        self.execution_environment.federate(
-            session_id, object_id, external_execution_environment_id, recursive
-        )
-
-    def unfederate_from_backend(self, dc_obj, external_execution_environment_id, recursive):
-        raise Exception("To refactor")
-        object_id = dc_obj._dc_meta.id
-        session_id = self.session.id
-        logger.debug(
-            "[==UnfederateObject==] Starting unfederation of object %s with ext backend %s, and session %s",
-            object_id,
-            external_execution_environment_id,
-            session_id,
-        )
-        self.execution_environment.unfederate(
-            session_id, object_id, external_execution_environment_id, recursive
-        )
+            await self.data_manager.flush_all()
+
+        # Stop metadata redis connection
+        await self.metadata_service.close()
+
+    # DEPRECATED
```

Comment:
This file has too many commented and deprecated lines. Explain in words, but non-functional non-descriptive lines should go away. (Reader/developer can explore git history).